### PR TITLE
[Feature] Support RTMDet-R deployment

### PR DESCRIPTION
**Pull request by liuyanyi**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Support RTMDet-R deployment in MMRotate dev-1.x

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: configs/mmrotate/rotated-detection_tensorrt-fp16_static-1024x1024.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+_base_ = [
+    './rotated-detection_static.py', '../_base_/backends/tensorrt-fp16.py'
+]
+
+onnx_config = dict(
+    output_names=['dets', 'labels'],
+    input_shape=(1024, 1024),
+    dynamic_axes={
```
Comment (grimoire) [inline configs/mmrotate/rotated-detection_tensorrt-fp16_static-1024x1024.py:8]: dynamic axes is not necessary for static config.

**Review by AllentDan — COMMENTED** [review]
May add unit tests and update docs.

**Comment by AllentDan** [comment]
May fix the lint following the [guide](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md#2-configure-pre-commit).

**Comment by liuyanyi** [comment]
> May fix the lint following the [guide](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md#2-configure-pre-commit).

It seems lint got an error because url dosn't exist on mmrotate 1.x branch, maybe i should change that to dev-1.x.
```
FILE: docs/en/04-supported-codebases/mmrotate.md
[✖] https://github.com/open-mmlab/mmrotate/blob/1.x/configs/rotated_rtmdet → Status: 404
```

**Review by grimoire — APPROVED** [review]
LGTM

**Review by AllentDan — APPROVED** [review]

**Pull request merged**